modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/actions/Actions.tsx

```diff
@@ -109,15 +98,15 @@ function Actions({actions, itemData, itemId, menuActive, onMenuActiveChange}) {
 
 		const url = formatActionURL(href, itemData);
 
-		const doAction = ({defaultPrevented}) => {
+		const doAction = ({defaultPrevented}: {defaultPrevented: boolean}) => {
 			if (target?.includes('modal')) {
 				event.preventDefault();
 
 				if (target === MODAL_PERMISSIONS) {
 					openPermissionsModal(url);
 				}
 				else {
-					openModal({
+					openModal!({
```

Comment:
These non-null assertions are kinda awkward. Would errors go away if we add `openModal`, `highlightItems` etc to `IFrontendDataSetContext`?

Reply:
It is already defined there https://github.com/liferay-frontend/liferay-portal/pull/3983/files#diff-82c0e084703354bbeba91216bc400b1c309d56e1695ec96b6205bedec0488a4bR39

Could not find a better solution :disappointed:

Reply:
All functions in `IFrontendDataSetContext` should almost certainly not be optional (`?`). It makes no sense, `openModal` will always be available. I am the one to blame. IIRC, my reasoning at the time was to put optional on everything, as the more permissive check, and figure out later on what must always be available. Not really the wisest line of reasoning in retrospective :sweat_smile:

Reply:
I'll make most of the functions required and remove the non-null assertions.

Reply:
![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/885d162e-e113-4a74-bbf6-e3c94b5cb52c)

:shrug:

Reply:
`executeAsyncItemAction: () => {},` I suppose. It does make sense that compiler requires an assignment in `createContext`. We know that there in only one [setting of context](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L839-L886), but in theory there could be others that don't include functions.